### ISSUE TITLE
a11y audit: ComboBox a11y fixes

### ISSUE THREAD

**Issue by marijohannessen** [carbon-design-system/carbon-components-react#603]
### Detailed description

- [ ] Each form control must have associated label (need to add `aria-label`)
<img width="1066" alt="screen shot 2018-02-12 at 2 46 14 pm" src="https://user-images.githubusercontent.com/5447411/36118674-81797ab0-1003-11e8-8728-57dc427bb787.png">
<img width="1042" alt="screen shot 2018-02-12 at 2 51 16 pm" src="https://user-images.githubusercontent.com/5447411/36118929-368a9768-1004-11e8-8395-1e8c2652b2fb.png">

- [ ] The WAI-ARIA role(s) and/or attribute(s) combobox, aria-autocomplete, aria-expanded are not valid for the element input
<img width="1076" alt="screen shot 2018-02-12 at 2 48 33 pm" src="https://user-images.githubusercontent.com/5447411/36118785-d2b06d4e-1003-11e8-9577-b24c8b01f2c6.png">

Ref https://github.ibm.com/Bluemix/carbon-issues/issues/496

**Comment by marijohannessen** [comment]
Hey @joshblack , after looking into this a bit further I think the second part is something that's being pulled in from Downshift right? So it's out of our control? The tool is complaining about the role/attibutes `combobox`, `aria-autocomplete` and `aria-expanded` not being valid on the `input` element.

![screen shot 2018-03-01 at 3 59 49 pm](https://user-images.githubusercontent.com/5447411/36872096-b7014f3c-1d69-11e8-8e3b-bfc6532fd9b5.png)

**Comment by joshblack** [comment]
@marijohannessen it might have been me mistakenly putting the prop getter on the wrong node 😬 definitely should be something we can fix!

**Comment by jblevins** [comment]
As the component currently exists, it fails accessibility testing.

Would it be possible to get the invalid aria-label which is rendered by this component removed? 

And would it be possible to get the other invalid aria attributes removed in Downshift?

**Comment by asudoh** [comment]
@dakahn Any thoughts? Does this component still fail AAT...?

**Comment by dakahn** [comment]
We're hamstrung by using Downshift here. Though the Downshift maintainers have[ recently mentioned ](https://github.com/downshift-js/downshift/issues/617)looking into their known accessibility concerns.

**Comment by joshblack** [comment]
@dakahn if we need to, we can definitely look into pulling downshift out 👍
